R/get_pvol_se.R

```diff
@@ -0,0 +1,43 @@
+get_pvol_se_radar_mapping <- c(
+  "seang" = "angelholm", "seatv" = "atvidaberg", "sebaa" = "balsta",
+  "sehem" = "hemse", "sehuv" = "hudiksvall", "sekaa" = "karlskrona",
+  "sekrn" = "kiruna", "selek" = "leksand", "sella" = "lulea",
+  "seoer" = "ornskoldsvik", "seosd" = "ostersund", "sevax" = "vara"
+)
+
+get_pvol_se <- function(radar, time, ..., call = rlang::caller_env()) {
+  if (!radar %in% names(get_pvol_se_radar_mapping)) {
+    cli::cli_abort(
+      c(
+        x = "The radar {.val {radar}} is not found in the mapping for Swedish radars",
+        i = "Most likely this means an invalid odim code has been provided"
+      ),
+      call = call, class = "getRad_error_get_pvol_se_radar_not_found"
+    )
+  }
+  radar_name <- get_pvol_se_radar_mapping[radar]
+  url <- glue::glue('/area/{radar_name}/product/qcvol/{lubridate::year(time)}/{lubridate::month(time)}/{{lubridate::day(time)}}/radar_{radar_name}_qcvol_{strftime(time, "%Y%m%d%H%M", tz="UTC" )}.h5')
+  req <- withCallingHandlers(
+    httr2::request(
+      "https://opendata-download-radar.smhi.se/api/version/latest"
+    ) |>
+      req_user_agent_getrad() |>
+      httr2::req_url_path_append(
+        glue::glue(url)
+      ) |>
+      httr2::req_perform(path = tempfile(fileext = ".h5"), error_call = call),
```

Comment:
Do we want to `req_retry_getrad()` or `req_cache_getrad()`?

Do we report progress when interactive for large files?

Reply:
I guess retries could be added, I guess I'm a bit more of the school of trying to avoid to many complications that do not solve real world problems that have been encountered :). For pipelines you would anyway implement some retries, a few hours later for error that are transient on other timescales